plasmapy/formulary/collisions.py

```diff
@@ -2033,3 +2056,209 @@ def coupling_parameter(
+        elif T_a is not None:
+            raise ValueError("Please specify either v_a or T_a, not both.")
+
+        self.test_particle = test_particle
+        self.field_particle = field_particle
+        self.v_a = v_a
+        self.T_a = (
+            T_a
+            if T_a is not None
+            else (0.5 * test_particle.mass * v_a**2 / k_B).to(u.K)
+        )
+        self.n_b = n_b
+        self.T_b = T_b
+        self.Coulomb_log = Coulomb_log
+
+        # These attributes are used in testing
+        self._x = self.x
+        self._v_0 = self.Lorentz_collision_frequency
+
+        phi = self._phi(self.x)
+        phi_prime = self._phi_prime(self.x)
+
+        mass_ratio = test_particle.mass / field_particle.mass
+
+        self.momentum_loss = (1 + mass_ratio) * phi * self._v_0
+        self.transverse_diffusion = (
+            2 * ((1 - 1 / (2 * self.x)) * phi + phi_prime) * self._v_0
+        )
+        self.parallel_diffusion = (phi / self.x) * self._v_0
+        self.energy_loss = 2 * (mass_ratio * phi - phi_prime) * self._v_0
+
+    @property
+    def Lorentz_collision_frequency(self):
+        r"""
+        The Lorentz collision frequency (see Ch. 5 of :cite:t:`chen:2016`) is given
+        by
+
+        .. math::
+
+            ν = n σ v \ln{Λ}
+
+        where :math:`n` is the particle density, :math:`σ` is the
+        collisional cross-section, :math:`v` is the inter-particle velocity
+        (typically taken as the thermal velocity), and :math:`\ln{Λ}` is the
+        Coulomb logarithm accounting for small angle collisions.
+
+        See Equation (2.14) in :cite:t:`callen:unpublished`.
+
+        The Lorentz collision frequency is equivalent to the variable
+        :math:`\nu_0^{\alpha/\beta}` on p. 31 of :cite:t:`nrlformulary:2019`.
+        """
+
+        return (
+            4
+            * np.pi
+            * (self.test_particle.charge_number * e.esu) ** 2
+            * (self.field_particle.charge_number * e.esu) ** 2
+            * self.Coulomb_log
+            * self.n_b
+            / (self.test_particle.mass**2 * self.v_a**3)
+        ).to(u.Hz)
+
+    @property
+    def x(self) -> u.dimensionless_unscaled:
+        """
+        The ratio of kinetic energy in the test particle to the thermal energy of the field particle.
+        This parameter determines the regime in which the collision falls (fast or slow).
+
+        """
+
+        x = self.field_particle.mass * self.v_a**2 / (2 * k_B.cgs * self.T_b)
+        return x.to(u.dimensionless_unscaled)
+
+    @staticmethod
+    def _phi_integrand(t: u.dimensionless_unscaled):
+        """
+        The phi integrand used in calculating phi
+        """
+
+        return t**0.5 * np.exp(-t)
+
+    def _phi_explicit(self, x: float) -> float:
+        """
+        The non-vectorized method for evaluating the integral for phi
+        """
+        integral, _ = scipy.integrate.quad(self._phi_integrand, 0, x)
```

Comment:
Maybe include a note in the documentation that this function uses `scipy.integrate.quad` with the default error tolerances to do the psi integral (which you should be adding to the documentation too). 

I hadn't seen this function before but it's clever: reduces the sensitivity to choosing an integration step size!

Reply:
Yeah the "vectorization" for calculating phi is just a for loop under the hood to ensure that we minimize the error as much as possible... I read somewhere that attempting to vectorize integration using other methods can cause unwanted error